homeassistant/components/zha/core/registries.py

```diff
@@ -121,22 +121,26 @@ def set_or_callable(value):
     return frozenset([str(value)])
 
 
+def _get_empty_frozenset() -> frozenset[str]:
+    return frozenset()
+
+
 @attr.s(frozen=True)
 class MatchRule:
     """Match a ZHA Entity to a channel name or generic id."""
 
-    channel_names: set[str] | str = attr.ib(
+    channel_names: frozenset[str] = attr.ib(
         factory=frozenset, converter=set_or_callable
     )
-    generic_ids: set[str] | str = attr.ib(factory=frozenset, converter=set_or_callable)
-    manufacturers: Callable | set[str] | str = attr.ib(
-        factory=frozenset, converter=set_or_callable
+    generic_ids: frozenset[str] = attr.ib(factory=frozenset, converter=set_or_callable)
+    manufacturers: frozenset[str] | Callable = attr.ib(
+        factory=_get_empty_frozenset, converter=set_or_callable
```

Comment:
We couldn't use `frozenset` here because `mypy` won't match with `frozenset[str] | Callable`

`Incompatible types in assignment (expression has type "FrozenSet[_T_co]", variable has type "Union[FrozenSet[str], Callable[..., Any]]")  [assignment]`

Using `_get_empty_frozenset` makes `mypy` happy.